### DISCUSSION TITLE
I can't figure out how to splice and delete an array element in a Store

### DISCUSSION THREAD

**Discussion by ni554n** [solidjs/solid#1788] [Q&A] [answered]
I've tried a bunch of ways to pinpoint the behavior but failed unfortunately. What am I missing?
https://playground.solidjs.com/anonymous/7a631161-032e-4722-95f3-7a0bd325da30

**Comment by atk** [comment]
We have [produce](https://www.solidjs.com/docs/latest#produce) for this use case.

**Comment by ni554n** [comment, marked as answer]
Oh, I was mutating the Store array directly. I should have copied it first then splice it.
